### ISSUE TITLE
Option to just output bash commands?

### ISSUE THREAD

**Issue by ewallace** [riboviz/RiboViz#63]
Would it be helpful for prepRiboViz to output a text file containing just all the shell commands into a text file, optionally without running them? That would mean that prepRiboViz and the yaml file work together as a shell script constructor.

This was motivated by reading the [documentation for RiboCode](https://github.com/xryanglab/RiboCode), which briefly attempts to explain how to set up these commands. Some users might want to construct the pipeline but then edit specific aspects themselves, or just to learn how the shell scripting works.

This might be made obsolete by moving to a workflow language.

**Comment by mikej888** [comment]
I think this would be a good idea. It could also help with debugging (both when running a pipeline, and for developers). It's analogous to the GNU Make `-n`, `--dry-run` option to show the commands that would be executed.

**Comment by FlicAnderson** [comment]
Will this output be effectively analagous to the documentation here: https://github.com/riboviz/RiboViz/blob/develop/docs/run-vignette.md#anatomy-of-prep_ribovizpy (barring details on outputs and descriptions)?

I'm certainly in favour of the idea, I think it would be quite handy.

**Comment by mikej888** [comment]
Yes, it'll be the commands that prep_riboviz.py sends on to bash.

**Comment by mikej888** [comment]
As of commit c5c6b6193305230b5c6f8c0fba7a0f91d3548034 `prep_riboviz.py` now creates a commands file, `prep_riboviz.sh`. The location/name can be changed by editing the `cmd_file` in `vignette_config.yaml`. 

This file contains all the bash commands needed to rerun the specific workflow, and can be run via:

```console
source prep_riboviz.sh
```

To create this file without actually running the commands, run `prep_riboviz.py` with a `--dry-run` parameter as its first argument:

```console
python -m riboviz.tools.prep_riboviz --dry-run rscripts/ vignette/vignette_config.yaml
```

**Comment by mikej888** [comment]
Merged via pull request #71.
